DevCycle/Networking/DevCycleService.swift

```diff
@@ -121,11 +121,11 @@ class DevCycleService: DevCycleServiceProtocol {
         var eventsRequest = createEventsRequest()
         let userEncoder = JSONEncoder()
         userEncoder.dateEncodingStrategy = .iso8601
-        guard let userId = user.userId, let userData = try? userEncoder.encode(user), let featureVariationMap = self.config.userConfig?.featureVariationMap else {
-            return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
+        guard let userId = user.userId, let userData = try? userEncoder.encode(user) else {
+            return completion((nil, nil, ClientError.MissingUser))
         }
-
-        let eventPayload = self.generateEventPayload(events, userId, featureVariationMap)
+        
+        let eventPayload = self.generateEventPayload(events, userId, self.config.userConfig?.featureVariationMap)
         guard let userBody = try? JSONSerialization.jsonObject(with: userData, options: .fragmentsAllowed) else {
             return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
```

Comment:
should we update this erro to be `ClientError.InvalidUser` since the `FeatureVariationMap` is optional now?